shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/presentation/modules/login/LoginViewModel.kt

```diff
@@ -0,0 +1,44 @@
+package co.nimblehq.blisskmmic.presentation.modules.login
+
+import co.nimblehq.blisskmmic.data.network.helpers.jsonApiException
+import co.nimblehq.blisskmmic.domain.model.Token
+import co.nimblehq.blisskmmic.domain.usecase.LogInUseCase
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.flow.*
+import kotlinx.coroutines.launch
+
+data class LoginViewState(
+    val isSuccess: Boolean = false,
+    val isLoading: Boolean = false,
+    val error: String? = null
+) {
+    @Suppress("EmptySecondaryConstructor")
+    constructor() : this(false, false, null) {}
+
+    @Suppress("EmptySecondaryConstructor")
+    constructor(error: String?) : this(false, false, error) {}
```

Comment:
Should we also add comments or TODOs for these empty constructors? 🤔

Reply:
These empty constructor will be used on the Swift side only because on Swift we can't call Constructor with default values. Do you think we need to add comments for all constructor for Swift?

Reply:
@blyscuit oh I see, it might be nice to have a comment here as a small reminder, but yeah if you think it is not needed, you can just ignore also, no biggie 👍